automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala

```diff
@@ -0,0 +1,56 @@
+package org.broadinstitute.dsde.firecloud.test.library
+
+import org.broadinstitute.dsde.firecloud.component.Component._
+import org.broadinstitute.dsde.firecloud.component.Label
+import org.broadinstitute.dsde.firecloud.config.UserPool
+import org.broadinstitute.dsde.firecloud.fixture.{UserFixtures, WorkspaceFixtures}
+import org.broadinstitute.dsde.firecloud.page.library.DataLibraryPage
+import org.broadinstitute.dsde.firecloud.test.{CleanUp, WebBrowserSpec}
+import org.scalatest.{FreeSpec, Matchers}
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
+
+class DataUseAwareSearchSpec extends FreeSpec with WebBrowserSpec with UserFixtures with WorkspaceFixtures with CleanUp with Matchers {
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  // We are only testing UI mechanics because the business logic of RP matching is extensively tested lower in the stack.
+  "Data Library" - {
+    "Open and close the RP modal" in withWebDriver { implicit driver =>
+      val user = UserPool.chooseAnyUser
+
+      withSignIn(user) { _ =>
+        val page = new DataLibraryPage().open
+
+        page.openResearchPurposeModal()
+
+        page.isShowingResearchPurposeModal shouldBe true
+
+        page.dismissResearchPurposeModal()
+
+        page.isShowingResearchPurposeModal shouldBe false
```

Comment:
another test case: for good measure, open and close it twice, to ensure that after closing it is re-openable. Think that's overkill?

Reply:
There's at least one more PR coming from this ticket, so I'll probably roll this into that one.